File: JavaScript/class/Composant.js
class Composant {
  constructor(position_x, position_y, hauteur, largeur, cheminImage) {
    this.position_x = position_x;
    this.position_y = position_y;
    this.hauteur = hauteur;
    this.largeur = largeur;
    this.image = new Image(this.hauteur, this.largeur);
    this.image.src = cheminImage;
  }

  getPosition_x() {
    return this.position_x;
  }

  setPosition_x(position_x) {
    this.position_x = position_x;
  }

  getPosition_y() {
    return this.position_y;
  }

  setPosition_y(position_y) {
    this.position_y = position_y;
  }

  getHauteur() {
    return this.hauteur ;
  }


  getLargeur() {
    return this.largeur ;
  }

  getImage() {
    return this.image ;
  }
}

File: Vipi_javascript_de_merde/src/Class/Composant.java
package Class;

import javax.swing.ImageIcon;

public class Composant {
	private Integer positionX;
	private Integer positionY;
	private Integer hauteur;
	private Integer largeur;
	private ImageIcon image;
	
	public Composant(Integer positionX, Integer positionY, Integer hauteur, Integer largeur, String linkImage) {
		this.positionX = positionX;
		this.positionY = positionY;
		this.hauteur = hauteur;
		this.largeur = largeur;
		this.image = new ImageIcon(getClass().getResource(linkImage));
	}
	
	public Integer getPositionX() {
		return positionX;
	}
	
	public void setPositionX(Integer positionX) {
		this.positionX = positionX;
	}
	
	public Integer getPositionY() {
		return positionY;
	}
	
	public void setPositionY(Integer positionY) {
		this.positionY = positionY;
	}
	
	public Integer getHauteur() {
		return hauteur;
	}
	
	public Integer getLargeur() {
		return largeur;
	}
	
	public ImageIcon getImage() {
		return image;
	}
	
	public boolean veri_touche (Vipi v) {
		int x = v.getPositionX();
		int y = v.getPositionY();
		if(x+v.getLargeur()> this.positionX && x < this.positionX + this.largeur) {
			if(y+v.getHauteur() < this.positionY) {
				return true ;
			}
		}
		return false ;
	}
	
}
